docs/viergewinnt.py

```diff
@@ -0,0 +1,439 @@
+
+    def gewonnen(self,board, player):
+        gew = False
+        # horizontal
+        for q in range(4):
+            for w in range(6):
+                if board[w][q] == player and board[w][q + 1] == player and board[w][q + 2] == player and board[w][q + 3] == player:
+                    gew = True
+        # vertikal
+        for q in range(7):
+            for w in range(3):
+                if board[w][q] == player and board[w + 1][q] == player and board[w + 2][q] == player and board[w + 3][q] == player:
+                    gew = True
+        # diagonal1
+        for q in range(4):
+            for w in range(3):
+                if board[w][q] == player and board[w + 1][q + 1] == player and board[w + 2][q + 2] == player and board[w + 3][q + 3] == player:
+                    gew = True
+        # diagonal2
+        for q in range(4):
+            for w in range(3):
+                if board[w][q + 3] == player and board[w + 1][q + 2] == player and board[w + 2][q + 1] == player and board[w + 3][q] == player:
+                    gew = True
+        return gew
+
+    def gameover(self,boar):
+        isover = True
+        for q in range(6):
+            if boar[q].count(' ') > 0:
+                isover = False
+        return isover
+
+    def play(self):
+        self.board = [
+        [' ', ' ', ' ', ' ', ' ', ' ', ' '],
+        [' ', ' ', ' ', ' ', ' ', ' ', ' '],
+        [' ', ' ', ' ', ' ', ' ', ' ', ' '],
+        [' ', ' ', ' ', ' ', ' ', ' ', ' '],
+        [' ', ' ', ' ', ' ', ' ', ' ', ' '],
+        [' ', ' ', ' ', ' ', ' ', ' ', ' '],
+        ]
+        #
+        self.turn=0
+        #
+        # Spieler:innen vorbereiten
+        # X muss immer zuerst spielen
+        self.players.clear()
+        self.players.append(ComputerPlayer('X'))
+        self.players.append(ComputerPlayer('O'))
+        #
+        current=0
+        while True:
+            print(self.turn)
+            self.printboard(self.board)
+            player = self.players[current]
+            print(player.token, ' ist am Zug')
+            self.board=player.get_move(copy.deepcopy(self.board))
+            current = (current + 1) % 2
+            self.turn+=1
+            if self.gameover(self.board) or self.gewonnen(self.board,'O')or self.gewonnen(self.board,'X'):
+                break
+        self.printboard(self.board)
+        if self.gewonnen(self.board,'O'):
+            print('O HAT GEWONNEN')
+        elif self.gewonnen(self.board,'X'):
+            print('X HAT GEWONNEN')
+        else:
+            print('UNENTSCHIEDEN')
+
+class Player(VierGewinnt):
+    def __init__(self, token):
+        super().__init__()
+        self.token = token
+
+    def fall(self, board, y, x, player):
+        if y <= 4:
+            if board[y + 1][x] == ' ':
+                board[y + 1][x] = player
+                board[y][x] = ' '
+                y = y + 1
+                self.fall(board, y, x, player)
+            else:
+                pass
```

Comment:
@luiscarlosgarzacisneros wozu wird der "player" hier übergeben? Mmn. geht es ohne, da das Spieler-Objekt sich selber kennt:
```suggestion
    def fall(self, board, y, x):
        if y <= 4:
            if board[y + 1][x] == ' ':
                board[y + 1][x] = self.token
                board[y][x] = ' '
                y = y + 1
                self.fall(board, y, x)
            else:
                pass
```